### PR TITLE
Explicitly declare and restrict sh dependency to < 2.0.0, and also declare other dependencies that is directly imported

### PR DESCRIPTION
**Pull request by felixfontein**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -42,6 +42,12 @@ jinja2 = "*"
 packaging = "*"
 rstcheck = ">= 3.0.0, < 7.0.0"
 sphinx = "*"
+sh = ">= 1.0.0 < 2.0.0"
```
Comment (felixfontein) [inline pyproject.toml:45]:
```suggestion
# sh v2 has breaking changes.
# https://github.com/ansible-community/antsibull-core/issues/34
sh = ">= 1.0.0 < 2.0.0"
```

Comment (gotmax23) [reply pyproject.toml:47]: Thanks

File: pyproject.toml [hunk 2]
```diff
@@ -42,6 +42,14 @@ jinja2 = "*"
 packaging = "*"
 rstcheck = ">= 3.0.0, < 7.0.0"
 sphinx = "*"
+# sh v2 has breaking changes.
+# https://github.com/ansible-community/antsibull-core/issues/34
+sh = ">= 1.0.0 < 2.0.0"
+pydantic = "*"
```
Comment (gotmax23) [inline pyproject.toml:48]: pydantic is the subject of a major rewrite. We might want to pin this to `< 2.0.0`.

Comment (gotmax23) [reply pyproject.toml:48]: In any case, this constraint should be the same between antsibull-docs and and antsibull-core.

**Review by gotmax23 — APPROVED** [review]

File: pyproject.toml (hunk 2 above)
Comment (felixfontein) [reply pyproject.toml:48]: Sounds like a good idea. I'll create a follow-up for that (later today or tomorrow).

**Comment by felixfontein** [comment]
@gotmax23 thanks for reviewing this!

File: pyproject.toml (hunk 2 above)
Comment (felixfontein) [reply pyproject.toml:48]: One annoying thing is that pydantic does not adhere to semver, so `< 2.0.0` does not prevent breaking changes. But restricting this more is not a good idea either, IMO...

**Pull request merged**